src/node/node_state.h

```diff
@@ -1595,68 +1596,50 @@ namespace ccf
       return last_recovered_signed_idx;
     }
 
-    void setup_recovery_hook(bool from_snapshot)
+    void setup_recovery_hook()
     {
-      // When recoverying from a snapshot, the first secret is valid from the
-      // version at which it was recorded
-      static bool is_first_secret = !from_snapshot;
-
       network.tables->set_map_hook(
-        network.shares.get_name(),
-        network.shares.wrap_map_hook(
-          [this](kv::Version version, const Shares::Write& w)
+        network.encrypted_ledger_secrets.get_name(),
+        network.encrypted_ledger_secrets.wrap_map_hook(
+          [this](
+            kv::Version version, const EncryptedLedgerSecretsInfo::Write& w)
             -> kv::ConsensusHookPtr {
-            for (const auto& [k, opt_v] : w)
+            if (w.size() > 1)
             {
-              if (!opt_v.has_value())
-              {
-                throw std::logic_error(
-                  fmt::format("Unexpected: removal from shares table ({})", k));
-              }
+              throw std::logic_error(fmt::format(
+                "Unexpected: multiple writes to {} table",
```

Comment:
I think I would prefer "Transaction contains {} writes to map {}, expected one".